entity-management-web/src/main/java/eu/europeana/entitymanagement/web/EMController.java

```diff
@@ -1,32 +1,109 @@
-
-    private static final String MY_REGEX = "^[a-zA-Z0-9_]*$";
-    private static final String INVALID_REQUEST_MESSAGE = "Invalid parameter.";
-
-    /**
-     * Test endpoint
-     * @param someRequest an alpha-numerical String
-     * @return just something, doesn't matter what
-     */
-    @GetMapping(value = "/{someRequest}", produces = MediaType.APPLICATION_JSON_VALUE)
-    public String handleMyApiRequest(
-        @PathVariable(value = "someRequest")
-            @Pattern(regexp = MY_REGEX, message = INVALID_REQUEST_MESSAGE) String someRequest) {
-        return "It works!";
-    }
+public class EMController extends BaseRest {
+	
+	@Autowired
+	EntityRecordService entityRecordService;
+    
+	@ApiOperation(value = "Retrieve a known entity", nickname = "getEntityJsonLd", response = java.lang.Void.class)
+	@RequestMapping(value = {"/entity/{type}/{namespace}/{identifier}.jsonld"}, method = RequestMethod.GET,
+			produces = {HttpHeaders.CONTENT_TYPE_JSONLD, MediaType.APPLICATION_JSON_VALUE})
+	public ResponseEntity<String> getJsonLdEntity(
+			@RequestParam(value = CommonApiConstants.PARAM_WSKEY, required=false) String wskey,
+			@PathVariable(value = WebEntityConstants.PATH_PARAM_TYPE) String type,
+			@PathVariable(value = WebEntityConstants.PATH_PARAM_NAMESPACE) String namespace,
+			@PathVariable(value = WebEntityConstants.PATH_PARAM_IDENTIFIER) String identifier,
+			HttpServletRequest request
+			) throws HttpException  {
+	    return createResponse(type, namespace, identifier, FormatTypes.jsonld, null, request);	
+	    
+	}
+	
+	@ApiOperation(value = "Retrieve a known entity", nickname = "getEntityXml", response = java.lang.Void.class)
+	@RequestMapping(value = {"/entity/{type}/{namespace}/{identifier}.xml"}, method = RequestMethod.GET, 
+		produces = {HttpHeaders.CONTENT_TYPE_APPLICATION_RDF_XML, HttpHeaders.CONTENT_TYPE_RDF_XML, MediaType.APPLICATION_XML_VALUE})
+	public ResponseEntity<String> getXmlEntity(
+			@RequestParam(value = CommonApiConstants.PARAM_WSKEY, required=false) String wskey,
+			@PathVariable(value = WebEntityConstants.PATH_PARAM_TYPE) String type,
+			@PathVariable(value = WebEntityConstants.PATH_PARAM_NAMESPACE) String namespace,
+			@PathVariable(value = WebEntityConstants.PATH_PARAM_IDENTIFIER) String identifier,
+			HttpServletRequest request
+			) throws HttpException  {
+	    return createResponse(type, namespace, identifier, FormatTypes.xml, HttpHeaders.CONTENT_TYPE_APPLICATION_RDF_XML, request);
+	}
+
+
+	private ResponseEntity<String> createResponse(String type, String namespace, String identifier, FormatTypes outFormat,  String contentType, HttpServletRequest request) throws HttpException{
+	    try {
+	    	
+	    	verifyReadAccess(request);
+	    	
+	    	EntityRecord entity = entityRecordService.retrieveEntityRecordByUri(type, namespace, identifier);
+	    	
+	    	String jsonLd = serialize(entity, outFormat);
+        
+    	    	Date timestamp = ((RankedEntity)entity.getEntity()).getTimestamp();
```

Comment:
the timestamp needs to be read from Aggregation object

Reply:
Ok.